codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/smoketests/SmokeTestsRunnerGenerator.kt

```diff
@@ -108,12 +128,72 @@ class SmokeTestsRunnerGenerator(
         writer.withInlineBlock("#L {", "}", service) {
             if (testCase.vendorParams.isPresent) {
                 testCase.vendorParams.get().members.forEach { vendorParam ->
-                    if (vendorParam.key.value == "region") {
-                        writeInline("#L = ", vendorParam.key.value.toCamelCase())
-                        declareSection(SmokeTestRegionDefault)
-                        write("#L", vendorParam.value.format())
-                    } else {
-                        write("#L = #L", vendorParam.key.value.toCamelCase(), vendorParam.value.format())
+                    when (vendorParam.key.value) {
+                        "region" -> {
+                            writeInline("#L = ", vendorParam.key.value.toCamelCase())
```

Comment:
style: repeated `vendorParam.key.value.toCamelCase()` can be stored in a local variable